lib/validators/contents.js

```diff
@@ -0,0 +1,96 @@
+const { Validator } = require('./validator')
+const minimatch = require('minimatch')
+const consolidateResult = require('./options_processor/options/lib/consolidateResults')
+const constructOutput = require('./options_processor/options/lib/constructOutput')
+
+const DEFAULT_SUCCESS_MESSAGE = 'Your Contents passes validations'
+const VALIDATOR_CONTEXT = { name: 'contents' }
+
+class Contents extends Validator {
+  constructor () {
+    super('contents')
+    this.supportedEvents = [
+      'pull_request.*',
+      'pull_request_review.*'
+    ]
+    this.supportedSettings = {
+      files: {
+        pr_diff: 'boolean',
+        ignore: 'array'
+      },
+      must_include: {
+        regex: 'string',
+        regex_flag: 'string',
+        message: 'string'
+      },
+      must_exclude: {
+        regex: 'string',
+        regex_flag: 'string',
+        message: 'string'
+      },
+      begins_with: {
+        match: ['string', 'array'],
+        message: 'string'
+      },
+      ends_with: {
+        match: ['string', 'array'],
+        message: 'string'
+      }
+    }
+  }
+
+  async validate (context, validationSettings) {
+    let fileOptions = null
+    let patternsToIgnore = ['.github/mergeable.yml']
+
+    let parseOption = validationSettings
+
+    if (validationSettings.files) {
+      fileOptions = validationSettings.files
+      if (fileOptions.ignore) patternsToIgnore = fileOptions.ignore
+      delete parseOption.files
+    }
+
+    let result = await context.github.pulls.listFiles(context.repo({pull_number: this.getPayload(context).number}))
+    let changedFiles = result.data
+      .filter(file => !matchesIgnoredPatterns(file.filename, patternsToIgnore))
+      .filter(file => file.status === 'modified' || file.status === 'added')
+      .map(file => file.filename)
+
+    let failedFiles = []
+    for (let file of changedFiles) {
+      const content = await getContent(context, file)
+
+      const processed = this.processOptions(parseOption, content)
```

Comment:
Will `processOption` work properly when content == null?

Reply:
if file we are fetching doesn't exist, the `getContent` will throw error. i don't think `null` can exist here

Reply:
You are returning null when it is a 404. 

ref: https://github.com/mergeability/mergeable/pull/318/files#diff-4c8888c6c6e37cf61921bb33ec55d224R90

Reply:
you are absolutely right, i missed that. I'll create a PR to fix that. 
In terms of severity, i think it's low because contents can only work with `pr_diff` ATM, which are files known to exist.